Added removeOverride() in rendition.themes to remove a css property in rendition.themes._overrides

strattonpress opened the pull request:
There are some instances that I want to remove and override a css property I have added using rendition.themes.override() and I can't seem to find a default method in epubjs to do this, so I added one. Maybe this will help some people too.

Usage:
`rendition.themes.removeOverride(cssProperty)`

strattonpress commented:
Or, we can do this instead:

`override(name, value, priority) {
    var contents = this.rendition.getContents();

    if(value) {
        this._overrides[name] = {
            value: value,
            priority: priority === true
        };
        
        contents.forEach((content) => {
            content.css(name, this._overrides[name].value, this._overrides[name].priority);
        });
    } else {
        delete this._overrides[name];
    }
}`

Which do you think is better?

fchasen commented:
Humm, I think I prefer explicitly removing the override.

Could you also add a removeOveride method to contents.js?

strattonpress commented:
Hi @fchasen !

Yes, I have already added a `removeOverride` method. The second one was just a suggestion.

Here it is: https://github.com/futurepress/epub.js/pull/971/commits/0acc4234aab439383243d23a4119bfbe879de9d5#diff-50c330c10da2c73d5d984ba2555b8e6e

In that method, I just removed the css property from _overrides then called the `css()` method from contents.js but without a css value (eg `css(property)`). Then in the `css()` method in contents.js, I did this:

`if (value) {
    content.style.setProperty(property, value, priority ? "important" : "");
} else {
    content.style.removeProperty(property);
}`

fchasen commented:
Sorry for the delay!

strattonpress commented:
Woah, thanks!

On Fri, 8 Nov 2019 at 01:44, Fred Chasen <notifications@github.com> wrote:

> Sorry for the delay!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/futurepress/epub.js/pull/971?email_source=notifications&email_token=ALCXBX67GE4X557XNSMLXS3QSRHXPA5CNFSM4IMK52RKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEDNHLGI#issuecomment-551187865>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALCXBX47Q7YF6O2RGMK3KYDQSRHXPANCNFSM4IMK52RA>
> .
>
